client/src/js/directives/bhUnique.js

```diff
@@ -49,7 +50,15 @@ function UniqueDirective($q, UniqueValidator) {
 
         UniqueValidator.check(validationUrl, viewValue)
           .then(function (valueExists) {
+            // Check if hospital_no change
+            var originUpdated = viewValue !== attrs.origin;
```

Comment:
This is pretty cool!

I think you might be able to do this without adding the `origin` part.  Could you try putting this at the top of your scope and seeing if it will do the same thing?
```js
var originValue = ctrl.$viewValue;

// now you can just compare to originValue
```
This will make it easier on programmers since they will not be obligated to put `origin` in the html.

Reply:
There is a problem, when initializing this variable, ctrl.$viewValue is empty

Reply:
Hmm... I wondered if that might happen... Okay, that is fine then.  We can leave it as it is for now. Thanks for trying!